### PR TITLE
Implemented unit conversion for real this time

### LINKED ISSUE #474: Actually convert the units from one to the other using the unit conversion formula
As a user I want to be able to change unit representation of variables, for example i want temperature variables represented in C or F or K as I desire. Furthermore, if one variable is in Farenhights and on is in Celcius, if I select Celcius as a unit, both of the variables are displayed in Celcius

**Description** 

**Additional Notes**

**Tasks**
- [ ] 
- [ ] 

**Acceptance Criteria**

### PR DESCRIPTION
**Pull request by ds161833**
Actually convert units from unit A to unit B by inversing conversionFormula

image
image
image

### CODE REVIEW

File: Server/src/services/DataUpload/EditUploadService.ts [hunk 1]
```diff
@@ -48,13 +48,6 @@ export default class EditUploadService extends AbstractUploadService {
         let arrayOfDatasetInfo = [this.datasetId, this.parsedFileData.dataset_name, dataSetDataTypeID, publicationID, subcategoryID, this.parsedFileData.data.comments]
         await this.insertDataset(this.uploadModel, arrayOfDatasetInfo)
 
-        // Run check on variable vs contents length to see if they're equal for data points and insert
-        if (this.parsedFileData.data.variables.length == this.parsedFileData.data.contents[0].point.length) {
```
Comment (Mehrdadmaskull) [inline Server/src/services/DataUpload/EditUploadService.ts:52]: Was there a reason this condition is removed?

**Review by Mehrdadmaskull — APPROVED** [review]
Nice and very clean code as usual and also nice refactoring. Other than my comment asking for an explanation, LGTM

**Review by silent-po2 — APPROVED** [review]
This looks really good! Works well on the page itself, nicely done

File: Server/src/services/DataUpload/EditUploadService.ts (hunk 1 above)
Comment (ds161833) [reply Server/src/services/DataUpload/EditUploadService.ts:52]: this condition was bugged and would throw, it should be fixable, but the case is unlikely, not a security threat, so better spend time on higher priority

**Pull request merged**